### ISSUE TITLE
VisDrone Dataset

### ISSUE THREAD

**Issue by cainiao123s** [ultralytics/ultralytics#22100]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I found that the VisDrone dataset cannot be trained using VisDrone.yaml. The file structure and annotation files of this dataset do not conform to the standard specifications.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @cainiao123s, thanks for opening this issue and for using Ultralytics 🚀. This is an automated response to help get things moving quickly — an Ultralytics engineer will follow up shortly.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Since your report concerns dataset structure and training with a dataset YAML, it may fall under a configuration/compatibility question. To help us assist you faster, please share:
- The exact command(s) you ran (CLI or Python) and `ultralytics` version
- A short log or error message output
- A small dataset tree snippet (folder and file layout) and 1–2 example annotation files
- Confirm you’re using the latest package and tried a simple sanity run (e.g., training a few epochs)

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/). A minimal training command for reference:
```bash
yolo train model=yolo11n.pt data=VisDrone.yaml imgsz=640 epochs=3
```

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
Thanks for the note—VisDrone ships with non‑YOLO labels; please use the maintained `VisDrone.yaml`, which auto‑downloads and converts the dataset to the correct `images/*` and `labels/*` layout as part of training (see the [VisDrone Dataset guide](https://docs.ultralytics.com/datasets/detect/visdrone/)).  
```bash
yolo detect train data=VisDrone.yaml model=yolo11n.pt imgsz=640
```  
If you previously downloaded/modified the dataset, delete that folder and rerun the command to trigger conversion; if it still fails, please paste the full traceback and the output of `yolo checks` for debugging.

**Comment by banu4prasad** [comment]
https://www.kaggle.com/datasets/banuprasadb/visdrone-dataset

**Comment by glenn-jocher** [comment]
Thanks—third‑party mirrors are fine as long as you let our converter run: unzip that Kaggle set into `datasets/VisDrone/VisDrone2019-DET-{train,val,test-dev}` and train with `data=VisDrone.yaml`; the converter in [VisDrone.yaml (source and conversion code)](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/cfg/datasets/VisDrone.yaml) will remap classes (1–10 → 0–9), skip ignored regions, and restructure to `images/*` and `labels/*`; if labels were edited, delete any `labels/**/*.cache` files and retry—if issues persist, please share the full traceback and `yolo checks` output.
